pkg/cassandra/nodetool/nodetool.go

```diff
@@ -0,0 +1,170 @@
+package nodetool
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/jetstack/navigator/pkg/cassandra/nodetool/client"
+)
+
+type NodeState string
+
+const (
+	NodeStateUnknown NodeState = "Unknown"
+	NodeStateNormal  NodeState = "Normal"
+	NodeStateLeaving NodeState = "Leaving"
+	NodeStateJoining NodeState = "Joining"
+	NodeStateMoving  NodeState = "Moving"
+)
+
+type NodeStatus string
+
+const (
+	NodeStatusUnknown NodeStatus = "Unknown"
+	NodeStatusUp      NodeStatus = "Up"
+	NodeStatusDown    NodeStatus = "Down"
+)
+
+type Node struct {
+	Host   string
+	ID     uuid.UUID
+	State  NodeState
+	Status NodeStatus
+	Local  bool
+}
+
+type NodeMap map[string]*Node
+
+func (nm NodeMap) LocalNode() *Node {
+	for _, node := range nm {
+		if node.Local {
+			return node
+		}
+	}
+	return nil
+}
+
+type Interface interface {
+	Status() (NodeMap, error)
+}
+
+type tool struct {
+	client client.Interface
```

Comment:
Ah, so the `nodetool` package depends upon the `client` package. It'd be great if we can get rid of this `tool` struct, and instead just have a `client` and then update the above `Interface` to include whatever is defined in `client`. Perhaps I'm missing something/perhaps I don't understand why you've done this in the first place.

Reply:
My idea was to separate out the Jolokia specific stuff into a separate package....it'd be better named `github.com/jetstack/navigator/pkg/cassandra/jolokia`.
* This would allow the interactions with Jolokia to be tested separately.
* And the `nodetool.Interface` could be tested using a fake implementation of `client.Interface` which returns canned responses.
* And if Jolokia doesn't work out, we could add a client that shells out to `nodetool` and parses its responses.

But I haven't got that far yet. I suggest I refactor in followup branches as I discover more about the requirements.

Reply:
👍 makes sense. I much prefer `github.com/jetstack/navigator/pkg/cassandra/jolokia` too. It's nice and obvious what a `github.com/jetstack/navigator/pkg/cassandra/jolokia.Client` does too (`jolokia.Client` is clearly a jolokia client 😄)